core/src/Kokkos_NumericTraits.hpp

```diff
@@ -35,9 +35,15 @@ namespace Kokkos::Experimental {
 namespace Impl {
 // clang-format off
 template <class> struct infinity_helper {};
+#if defined(_MSC_VER)
+template <> struct infinity_helper<float> { static constexpr float value = __builtin_huge_valf(); };
+template <> struct infinity_helper<double> { static constexpr double value = __builtin_huge_val(); };
+template <> struct infinity_helper<long double> { static constexpr long double value = __builtin_huge_val(); };
```

Comment:
Why don't you just use `std::numeric_limits` directly?

Reply:
This is a fair question.  In general we are not using it because it is not portable in kernels but I suppose it may be fine in the context of defining these compile-time constants.

I suppose I would prefer not have the extra dependency, more particularly on the one thing we mean to replace.

Feel free to experiment with it if you feel strongly about this.  I don't have time for it in the near term.

Reply:
https://godbolt.org/z/4h666Wsr6 seems to work for `Cuda`.

Reply:
Agreeing that the approach you suggest is superior.